src/engine/ChromaticFilter.cs

```diff
@@ -0,0 +1,34 @@
+using Godot;
+
+/// <summary>
+/// A chromatic aberration and barrel distrortion filter effect
+/// </summery>
+public class ChromaticFilter : TextureRect
+{
+    private static ChromaticFilter instance;
+    ShaderMaterial material;
+
+    public static ChromaticFilter Instance => instance;
+    public override void _Ready()
+    {
+        material = (ShaderMaterial) Material;
+        SetAmount(Settings.Instance.ChromaticAmount);
+        Show();
+    }
+
+    public void ToggleEffect(bool enabled)
```

Comment:
I think that this method is not really needed. I think that with the colourblind filters Show and Hide are directly called. So places calling this method should just directly call Show and Hide

Reply:
I need this here at the moment, because I'm setting up the scene in _Ready and just pulling the info from Settings.cs. It never has settings directly pushed to it, so nothing can directly call hide or show.